include/controller/c/webots/robot.h

```diff
@@ -65,7 +65,12 @@ int wb_robot_init_msvc();  // internally, this function just calls wb_robot_init
 #define wb_robot_init() (setvbuf(stdout, NULL, _IONBF, 0), setvbuf(stderr, NULL, _IONBF, 0), wb_robot_init_msvc())
 #endif
 
-int wb_robot_step(int duration);                                                                // milliseconds
+int wb_robot_step(int duration);  // milliseconds
+
+#ifdef __CYGWIN__  // in that case, we need to flush explicitly the stdout/stdin streams otherwise they are buffered
+#define wb_robot_step(d) (fflush(NULL) ? wb_robot_step(d) : wb_robot_step(d))
```

Comment:
I don't understand the reason to create a macro for this, rather than simply put the fflush statement into the wb_robot_step() function core. This line looks like a hack (difficult to understand and "hidden"). If there are reasons for this, a comment would be good.

Reply:
Sorry to insist, but in this case, wouldn't it be cleaner to build libController with gcc7 as we spoke yesterday?

Reply:
Good idea. Let me try that...

Reply:
For some reason, it doesn't work: the streams are not flushed when the libController and the controller are compiled with GCC7/cygwin. It seems GCC7/cygwin blocks the DLLs from being able to flush, regardless if the DLL was compiled with GCC7/cygwin or GCC8. So, I have no better idea than my original one.